src/worker.ts

```diff
@@ -1,11 +1,39 @@
 import "unfetch/polyfill";
 import { CONFIG_URL } from "./constants";
+import Fetch from "./tasks/fetch";
+import Pop from "./tasks/pop";
+import Task from "./tasks/task";
 
-interface Response {
-  json(): Promise<Config>;
+interface TaskLookup {
```

Comment:
My preference would be for this to be represented in a `src/tasks/index.ts` file which exports each task type. We then just import and the compiler should infer import types implicitly:

```js
import * as Tasks from './tasks';

function runTasks(config: Config) {
  const tasks = config.tasks.map((t): Task => new Tasks[t.type]);
} 
```